src/icon2svg.rs

```diff
@@ -0,0 +1,163 @@
+//! Produces svgs of icons in Google-style icon fonts
+
+use crate::{error::DrawSvgError, iconid::IconIdentifier};
+use kurbo::{Affine, BezPath, PathEl, Point};
+use skrifa::{
+    instance::{LocationRef, Size},
+    outline::DrawSettings,
+    raw::TableProvider,
+    FontRef, MetadataProvider,
+};
+use write_fonts::pens::{BezPathPen, TransformPen};
+
+fn round2(v: f64) -> f64 {
+    (v * 100.0).round() / 100.0
+}
+
+fn push_point(svg: &mut String, prefix: char, p: Point) {
+    svg.push(prefix);
+    svg.push_str(&format!("{},{}", round2(p.x), round2(p.y)));
+}
+
+/// We use this rather than [`BezPath::to_svg`]` so we can exactly match the output of the tool we seek to replace
+fn push_drawing_commands(svg: &mut String, path: &BezPath) {
+    for el in path.elements() {
+        match el {
+            PathEl::MoveTo(p) => push_point(svg, 'M', *p),
+            PathEl::LineTo(p) => push_point(svg, 'L', *p),
+            PathEl::QuadTo(p1, p2) => {
+                push_point(svg, 'Q', *p1);
+                push_point(svg, ' ', *p2);
+            }
+            PathEl::CurveTo(p1, p2, p3) => {
+                push_point(svg, 'C', *p1);
+                push_point(svg, ' ', *p2);
+                push_point(svg, ' ', *p3);
+            }
+            PathEl::ClosePath => svg.push('Z'),
+        }
+    }
+}
+
+pub fn draw_icon(font: &FontRef, options: &DrawOptions<'_>) -> Result<String, DrawSvgError> {
+    let upem = font
+        .head()
+        .map_err(|e| DrawSvgError::ReadError("cmap", e))?
+        .units_per_em();
+    let gid = options
+        .identifier
+        .resolve(font)
+        .map_err(|e| DrawSvgError::ResolutionError(options.identifier.clone(), e))?;
+
+    let glyph = font
+        .outline_glyphs()
+        .get(gid)
+        .ok_or(DrawSvgError::NoOutline(options.identifier.clone(), gid))?;
+
+    // Draw the glyph. Fonts are Y-up, svg Y-down so flip-y.
+    let mut path_pen = BezPathPen::new();
+    let mut transform_pen = TransformPen::new(&mut path_pen, Affine::FLIP_Y);
+
+    glyph
+        .draw(
+            DrawSettings::unhinted(Size::unscaled(), options.location),
+            &mut transform_pen,
+        )
+        .map_err(|e| DrawSvgError::DrawError(options.identifier.clone(), gid, e))?;
+
+    let upem_str = format!("{upem}");
+    let width_height = format!("{}", options.width_height);
```

Comment:
no functional difference but you can write both of these as `thing.to_string()`.

Reply:
Reads better to me, so changed.